src/wscearth/views.py

```diff
@@ -75,12 +98,22 @@ def positions():
     #    print(row)
 
     if len(df) == 0:
-        return flask.render_template("positions_map.html",
-                                    centre_lat=-25.0,
-                                    centre_long=130.0,
-                                    df=[])
-
-    return flask.render_template("positions_map.html",
-                                    centre_lat=df["latitude"].mean(),
-                                    centre_long=df["longitude"].mean(),
-                                    df=df)
+        df = []
+        center = {
+            "lat": -25.0,
+            "lng": 130.0,
+        }
+    else:
+        center = {
+            "lat": df["latitude"].mean(),
+            "lng": df["longitude"].mean(),
+        }
+
+    # We're in 2023 and this the best Python can do?
+    items = json.loads(df.to_json(orient="records"))
```

Comment:
nit: I think we could do df.to_dict("records") here?

Reply:
Afraid not. The items within the dictionary are still DataFrames which aren't JSON serialisable.

In the end this is probably still faster (and more robust): `C-ext -> string -> C-ext -> dict` instead of a lump of python recursive loops.

Reply:
Ooh,  what I did miss was on line 101 `df` becomes a list which has no `.to_json()`. So I suppose this should all look like this:

```py
items = []
center = { "lat": -25.0, "lng": 130.0 }

if len(df) > 0:
  items = json.loads(df.to_json(orient="records"))
  center = {
    "lat": df["latitude"].mean(),
    "lng": df["longitude"].mean(),
  }
```